File: app/src/main/java/com/zhenl/crawler/MyApplication.java
package com.zhenl.crawler;

import android.app.Application;

/**
 * Created by lin on 2018/8/23.
 */
public class MyApplication extends Application {

    public static MyApplication application;

    @Override
    public void onCreate() {
        super.onCreate();
        application = this;
    }

    public static MyApplication getInstance() {
        return application;
    }
}

File: app/src/main/java/com/zhenl/crawler/MyApplication.kt
package com.zhenl.crawler

import android.app.Application
import androidx.lifecycle.MutableLiveData
import com.zhenl.violet.core.ResMgr

/**
 * Created by lin on 2018/8/23.
 */
class MyApplication : Application() {

    companion object {
        lateinit var instance: MyApplication
        val globalLoading = MutableLiveData<Boolean>()
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
        ResMgr.init(this)
    }
}